doc/dev/sharp_bits.rst

```diff
@@ -0,0 +1,1015 @@
+...     qml.RX(x, wires=0)
+...     return {"samples": qml.sample(), "expval": qml.expval(qml.PauliZ(0))}
+>>> circuit(0.3)
+{'expval': array(-0.9899925),
+ 'samples': array([[1., 0.],
+        [1., 0.],
+        [1., 0.],
+        [1., 0.],
+        [1., 0.],
+        [1., 0.],
+        [1., 0.],
+        [0., 0.],
+        [1., 0.],
+        [1., 0.]])}
+
+
+Recursion
+---------
+
+Recursion is not currently supported, and will result in errors. For example,
+
+.. code-block:: python
+
+    @qjit(autograph=True)
+    def fibonacci(n: int):
+        if n <= 1:
+            return n
+        return fibonacci(n-1) + fibonacci(n-2)
+
+>>> fibonacci(10)
+RecursionError: maximum recursion depth exceeded in comparison
+
+
+This is due to the fact that during compilation, Catalyst tries to evaluate
+both branches of the conditional statement recursively; because there is
+``n`` is a dynamic variable, it has no concrete value at compile time, and
+tracing can never complete.
+
+Instead, try to write your program without recursion. For example, in this case
+we can use a while loop:
+
+.. code-block:: python
+
+    @qjit
+    def fibonacci(n):
+
+        @catalyst.while_loop(lambda count, *args: count < n)
+        def loop_fn(count, a, b, sum):
+            a, b = b, sum
+            sum = a + b
+            return count + 1, a, b, sum
+
+        _, _, _, result  = loop_fn(1, 0, 1, 1)
+        return result
+
+>>> fibonacci(10)
+array(89)
+
+Compatibility with broadcasting
+-------------------------------
+
+Catalyst does not currently support passing multi-dimensional arrays
+as quantum operator parameters ('parameter broadcasting'):
+
+>>> @qml.qnode(dev)
+... def circuit(x):
+...     qml.RX(x, wires=0)
+...     qml.RY(0.1, wires=0)
+...     return qml.expval(qml.PauliZ(0))
+>>> circuit(jnp.array([0.1, 0.2]))
+Array([0.99003329, 0.97517033], dtype=float64)
+>>> qjit(circuit)(jnp.array([0.1, 0.2]))
+UnboundLocalError: local variable 'baseType' referenced before assignment
+
+While not as flexible as true vectorized quantum operations, as a workaround
+``jax.vmap`` can be used to allow for multi-dimensional **function**
+arguments:
+
+>>> jax.vmap(qjit(circuit))(jnp.array([0.1, 0.2]))
+Array([0.99003329, 0.97517033], dtype=float64)
+
+Note that ``jax.vmap`` cannot be used within a qjit-compiled function:
+
+>>> qjit(jax.vmap(circuit))(jnp.array([0.1, 0.2]))
+NotImplementedError: Batching rule for 'qinst' not implemented
+
+Functionality differences from PennyLane
+----------------------------------------
+
+The ultimate aim with Catalyst will be the ability to prototype quantum algorithms
+in Python with PennyLane, and easily scale up prototypes by simply adding ``@qjit``.
+This will require that all PennyLane functionality behaves identically whether or not
+the ``@qjit`` decorator is applied.
+
+Currently, however, this is not the case in two places.
+
+- **Finite-shot measurement statistics**. The Catalyst-Lightning runtime does not
```

Comment:
Hmm.. The Catalyst Runtime has support for finite-shot measurement statistics, but the Lightning simulators don't provide this support in their C++ interface and as the result this feature is not supported in Catalyst.  

```suggestion
- **Finite-shot measurement statistics**. The Catalyst Runtime does not
```

Reply:
@maliasadi I think I would prefer to keep `Lightning` here, since as you say, this is lightning that does not support it!